hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java

```diff
@@ -218,17 +252,18 @@ public boolean apply(ComputeMetadata nodeMetadata) {
                     return false;
                 }
                 for (AbstractMap.SimpleImmutableEntry entry: tagPairs) {
-                    if (!nodeMetadata.getUserMetadata().get(entry.getKey()).
-                            equals(entry.getValue())) {
+                    String value = nodeMetadata.getUserMetadata().get(entry.getKey());
+                    if (value != null && equals(entry.getValue())) {
```

Comment:
Good catch also method should return true if tag key/value pair matches. Corrected.